### ISSUE TITLE
Faster RCNN laid out in Paper not in Repo

### ISSUE THREAD

**Issue by elnazsn1988** [microsoft/unilm#249]
Hi - In the paper shared for this model, the layoutlm architecture is used in tandem with 2 seperate entities, an HTR or OCR model for text extraction for images, to be fed into the model for word level classification (Question, Answer other etc) , with an additional classification needed to be able to extract question and answers from the documents. Additionally, a Faster-RCNN is shown to be used in tandem with the outputs from the layoutlm, the presence/usage of which is not clear :

![image](https://user-images.githubusercontent.com/36118289/94671255-9092de80-030b-11eb-87c5-fff0ba11a958.png)

 My question is if the Faster RCNN is used anywhere in this repo and if so, where does it reside?

**Comment by vineethcmps** [comment]
Yep couldn't find the Image embedding used at all.

**Comment by elnazsn1988** [comment]
@vineethcmps my guess is its not included in the repo, however it is probably the FCNN referenced in the original FUNSD paper, i.e. the Layoutlm embeddings and the embeddings from the FCNN can be used together, how though I dont quite understand.

**Comment by w-cheng** [comment]
@elnazsn1988 You're right that the Faster RCNN mentioned in the paper is not in this repo. You can use any off-the-shelf Faster RCNN model to combine output from LayoutLM model. See my issue here: #163 

You probably need to add an additional Dense layer to make the feature dimensions match.

**Comment by elnazsn1988** [comment]
> @elnazsn1988 You're right that the Faster RCNN mentioned in the paper is not in this repo. You can use any off-the-shelf Faster RCNN model to combine output from LayoutLM model. See my issue here: #163
> 
> You probably need to add an additional Dense layer to make the feature dimensions match.

Hey @w-cheng Thanks very much for your reply, I suppose my issue was the FUNSD paper themselves trained a Faster RCNN using pytorch, what I wasn't clear on was how the question-answer etc annotation would work in pytorch, and how the FUNSD dataset can be edited to be used as a training set for a Faster RCNN on pytorch or elsewhere. Do you have any suggestions on tutorials/Repo's to look at? any and all help would be much appreciated.

**Comment by w-cheng** [comment]
@elnazsn1988 You're very welcome. From what I understand, the FUNSD paper trained a text object detector using Mask-RCNN. 

The training dataset required for Mask-RCNN are the classes and the bounding boxes associated with each object in every image. 

In the case of FUNSD dataset, there would be two classes: text and background, the bounding box of each text object is the `box` element in the json file. I think you need to write a script to convert them into a suitable format for whichever Mask-RCNN implementation you choose.

In terms of tutorials/repos, this might be useful: https://github.com/matterport/Mask_RCNN, it talks about training Mask RCNN on custom training dataset. 

The toy shape dataset example would be a good starting point to understand the training procedure: https://github.com/matterport/Mask_RCNN/blob/master/samples/shapes/train_shapes.ipynb
